pom.xml

```diff
@@ -18,7 +18,7 @@
 
   <properties>
     <java.version>17</java.version>
-    <openapi-generator.version>6.2.1</openapi-generator.version>
```

Comment:
Proper superclass/inheritance was introduced in openapi-generated 6.5.0/6.6.0 (parameter `REF_AS_PARENT_IN_ALLOF`); 6.6.0 is the latest version not yet on v7, so I figured we should go ahead and upgrade to that.

See https://github.com/OpenAPITools/openapi-generator/issues/10010 and https://github.com/OpenAPITools/openapi-generator/pull/15039 for more information